homeassistant/components/incomfort/config_flow.py

```diff
@@ -63,9 +78,60 @@ async def async_try_connect_gateway(
     return None
 
 
+class InComfortOptionsFlowHandler(OptionsFlow):
+    """Handle InComfort Lan2RF gateway options."""
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Initiate options flow."""
+        return await self.async_step_gateway()
+
+    async def async_step_gateway(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Manage the options."""
+        errors: dict[str, str] | None = None
+        if (
+            user_input is not None
+            and (
+                errors := await async_try_connect_gateway(
+                    self.hass, dict(self.config_entry.data)
+                )
+            )
+            is None
+        ):
+            new_options: dict[str, Any] = self.config_entry.options | user_input
+            self.hass.config_entries.async_update_entry(
+                self.config_entry, options=new_options
+            )
+            self.hass.async_create_task(
+                self.hass.config_entries.async_reload(self.config_entry.entry_id),
+                eager_start=False,
+            )
+            return self.async_create_entry(data=new_options)
```

Comment:
I know we have this helper for reauth, does it work for options flow as well?

Reply:
Oh I didn't mean the reload helper, I meant the update data and reload integration one

Reply:
I know, but this was another improvement

Reply:
As we want to class create entry, we do not want to reload and abort, as we do with reauth in the config flow.